storage/factory/common.go

```diff
@@ -117,3 +134,77 @@ func getLastEpochFromDirNames(epochDirs []string) (uint32, error) {
 
 	return epochsInDirName[0], nil
 }
+
+func getLastEpochFromStorage(
```

Comment:
tests?

Reply:
pretty hard to do that, I have to agree. However, this is a temporary solution until we create a static component which holds all the needed data

Reply:
I know, in order for the tests to be written in an easier way, the file/directory specific functions can be extracted in functions that can be mocked in tests. In this way we can reach as much as 100% code coverage, effectively testing each branch in each situation. Can be done in a future PR.